### PR TITLE
NAS-102286 / 11.3 /  Adapt iocage to abstract git repo for plugins

### PR DESCRIPTION
**Pull request by sonicaj**
This PR introduces changes to make sure that iocage can now work with multiple git repos for fetching/updating plugins.

### CODE REVIEW

**Review by skarekrow — DISMISSED** [review]

**Comment by skarekrow** [comment]
Good job!

File: iocage_lib/ioc_check.py [hunk 1]
```diff
@@ -42,21 +43,32 @@ def __init__(self, silent=False, callback=None, migrate=False):
         self.callback = callback
         self.silent = silent
 
+        self.zfs = iocage_lib.ioc_json.IOCZFS()
+        self.pool_mountpoint = self.zfs.zfs_get_property(
+            self.pool, 'mountpoint'
+        )
+        self.iocage_mountpoint = self.zfs.zfs_get_property(
+            f'{self.pool}/iocage', 'mountpoint'
```
Comment (william-gr) [inline iocage_lib/ioc_check.py:54]: Doesnt this need to be after check datasets?

Comment (sonicaj) [reply iocage_lib/ioc_check.py:54]: Oh, yes it indeed should be. Thank you for pointing it out.

File: iocage_lib/ioc_plugin.py [hunk 2]
```diff
@@ -76,6 +77,26 @@ def __init__(self, release=None, jail=None, plugin=None, branch=None,
...
+                    'message': f'Git Repository URI is invalid.'
+                },
+                _callback=self.callback
+            )
+
+        self.git_destination = kwargs.get('git_destination')
+        if not self.git_destination:
+            self.git_destination = os.path.join(
+                self.iocroot, '.plugins', '_'.join(
+                    self.git_repository.split('/')[-2:]
```
Comment (william-gr) [inline iocage_lib/ioc_plugin.py]: http://mygitrepo.com/somefolder/foo/bar could happen (not on github). rsplit('/', 1)[-1]  for safety?

**Review by william-gr — APPROVED** [review]

**Pull request merged**